include/pybind11/attr.h

```diff
@@ -185,14 +185,20 @@ struct function_record {
     /// True if the function has a '**kwargs' argument
     bool has_kwargs : 1;
 
-    /// True once a 'py::kwonly' is encountered (any following args are keyword-only)
-    bool has_kwonly_args : 1;
+    /// True once a 'py::kw_only' is encountered (any following args are keyword-only)
+    bool has_kw_only_args : 1;
+
+    /// True once a 'py::pos_only' is encountered (any previous args are pos-only)
+    bool has_pos_only_args : 1;
 
     /// Number of arguments (including py::args and/or py::kwargs, if present)
     std::uint16_t nargs;
 
     /// Number of trailing arguments (counted in `nargs`) that are keyword-only
-    std::uint16_t nargs_kwonly = 0;
+    std::uint16_t nargs_kw_only = 0;
+
+    /// Number of trailing arguments (counted in `nargs`) that are positional-only
```

Comment:
`kw_only` is more common and was introduced (in normal Python syntax) first, so I rather like leading with it.

Reply:
OK, that also makes sense!

Do we need to clarify somewhere that this works pre-Python 3.8 as well?

Reply:
It's in the docs:

```diff
+Positional-only arguments
+=========================
+
+Python 3.8 introduced a new positional-only argument syntax, using ``/`` in the
+function definition (note that this has been a convention for CPython
+positional arguments, such as in ``pow()``, since Python 2). You can
+do the same thing in any version of Python using ``py::pos_only()``:
```

Reply:
Sorry! Overlooked that.